engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EditFileCmd.scala

```diff
@@ -23,10 +24,13 @@ class EditFileCmd(request: Api.EditFileNotification) extends Command(None) {
     ctx.locking.acquireFileLock(request.path)
     ctx.locking.acquirePendingEditsLock()
     try {
-      ctx.state.pendingEdits.enqueue(request.path, request.edits)
+      val edits = request.edits.map(edit => PendingEdit(edit, request.execute))
+      ctx.state.pendingEdits.enqueue(request.path, edits)
```

Comment:
Rather then introducing `PendingEdit` one could have the original pending `TextEdit` and just a `Boolean` flag whether execution is needed. The processing of pending edits always is:
- apply all of them
- if one of them requested execution, perform execution at the end

E.g. we could got without `PendingEdit` class, I think.

Reply:
I don't see an issue with creating extra type. Imo, execute flag does not belong to the `TextEdit`, it's something external. Another reason I can think of, `TextEdit` and `PendingEdit` have different meaning and it's not a bad idea to create different types for them